algorithms/src/sorting/Kokkos_Sort.hpp

```diff
@@ -0,0 +1,196 @@
+#include <algorithm>
+
+namespace Kokkos {
+// clang-format off
+
+//
+// basic overloads
+//
+template <class ExecutionSpace, class DataType, class... Properties>
+void sort([[maybe_unused]] const ExecutionSpace& exec,
+          const Kokkos::View<DataType, Properties...>& view)
+{
+  // constraints
+  using ViewType = Kokkos::View<DataType, Properties...>;
+  using MemSpace = typename ViewType::memory_space;
+  static_assert( (ViewType::rank == 1),
+		 "Kokkos::sort: currently supporting 1D Views only.");
+  static_assert(SpaceAccessibility<ExecutionSpace, MemSpace>::accessible,
+		"Kokkos::sort: execution space instance is not able to access the memory space of the "
+		"View argument!");
+
+  // trivial cases
+  if (view.extent(0) <= 1) { return; }
+
+  // non trivial
+  if constexpr (SpaceAccessibility<HostSpace, MemSpace>::accessible) {
+    auto first = ::Kokkos::Experimental::begin(view);
+    auto last  = ::Kokkos::Experimental::end(view);
+    std::sort(first, last);
+  } else {
+    Impl::sort_device_view_without_comparator_and_fence_exec(exec, view);
+  }
+}
+
+template <class DataType, class... Properties>
+void sort(const Kokkos::View<DataType, Properties...>& view)
+{
+  // constraints
+  using ViewType = Kokkos::View<DataType, Properties...>;
+  static_assert( (ViewType::rank == 1),
+		 "Kokkos::sort: currently supporting 1D Views only.");
+
+  // trivial cases
+  if (view.extent(0) <= 1) { return; }
+
+  // non trivial
+  using ViewType = Kokkos::View<DataType, Properties...>;
+  typename ViewType::execution_space exec;
+  sort(exec, view);
+  exec.fence("Kokkos::sort: fence after sorting");
+}
+
+//
+// overloads accepting a comparator
+//
+template <class ExecutionSpace, class CompType, class DataType,
+          class... Properties>
+void sort([[maybe_unused]] const ExecutionSpace& exec,
+          const Kokkos::View<DataType, Properties...>& view,
+          CompType const& comp)
+{
+  // constraints
+  using ViewType = Kokkos::View<DataType, Properties...>;
+  using MemSpace = typename ViewType::memory_space;
+  static_assert(
+      SpaceAccessibility<ExecutionSpace, MemSpace>::accessible,
+      "Kokkos::sort: execution space instance is not able to access the memory space of the "
+      "View argument!");
+  static_assert(
+      ViewType::rank == 1 &&
+          (std::is_same_v<typename ViewType::array_layout, LayoutRight> ||
+           std::is_same_v<typename ViewType::array_layout, LayoutLeft> ||
+	   std::is_same_v<typename ViewType::array_layout, LayoutStride>),
+      "Kokkos::sort: supports 1D Views with LayoutRight, LayoutLeft or LayoutStride.");
+
+  // trivial cases
+  if (view.extent(0) <= 1) { return; }
+
+  if constexpr (SpaceAccessibility<HostSpace, MemSpace>::accessible) {
+    auto first = ::Kokkos::Experimental::begin(view);
+    auto last  = ::Kokkos::Experimental::end(view);
+    std::sort(first, last, comp);
+  } else {
+    Impl::sort_device_view_with_comparator_and_fence_exec(exec, view, comp);
```

Comment:
Why "and fence"?  The overload that takes an execution space as 1st argument is supposed to be asynchronous.

Reply:
my bad, i was mislead by https://kokkos.github.io/kokkos-core-wiki/API/algorithms/Sort.html#additional-information

Reply:
I will fix the fencing where needed

Reply:
Yes